apps/console/src/features/identity-providers/components/settings/connected-apps.tsx

```diff
@@ -0,0 +1,529 @@
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+
+import { isFeatureEnabled } from "@wso2is/core/helpers";
+import { AlertLevels, TestableComponentInterface } from "@wso2is/core/models";
+import { addAlert } from "@wso2is/core/store";
+import { 
+    AnimatedAvatar, 
+    AppAvatar,  
+    DataTable, 
+    EmphasizedSegment,
+    EmptyPlaceholder, 
+    Heading,
+    TableActionsInterface, 
+    TableColumnInterface 
+} from "@wso2is/react-components";
+import React, 
+{ 
+    FunctionComponent,
+    ReactElement, 
+    ReactNode, 
+    SyntheticEvent, 
+    useEffect, 
+    useState 
+} from "react";
+import { useTranslation } from "react-i18next";
+import { useDispatch, useSelector } from "react-redux";
+import {  
+    Divider,
+    Header, 
+    Icon, 
+    Input, 
+    Label, 
+    SemanticICONS
+} from "semantic-ui-react";
+import { applicationListConfig } from "../../../../extensions/configs/application-list";
+import { getApplicationDetails } from "../../../applications/api";
+import { ApplicationManagementConstants } from "../../../applications/constants";
+import { 
+    ApplicationAccessTypes, 
+    ApplicationBasicInterface, 
+    ApplicationListItemInterface, 
+    ApplicationTemplateListItemInterface 
+} from "../../../applications/models";
+import { ApplicationTemplateManagementUtils } from "../../../applications/utils";
+import {  
+    AppConstants,
+    AppState, 
+    FeatureConfigInterface,   
+    UIConstants, 
+    getEmptyPlaceholderIllustrations, 
+    history
+} from "../../../core";
+import { getIDPConnectedApps } from "../../api";
+import { 
+    ConnectedAppInterface,
+    ConnectedAppsInterface, 
+    IdentityProviderInterface 
+} from "../../models";
+
+
+/**
+ * Proptypes for the advance settings component.
+ */
+interface ConnectedAppsPropsInterface extends TestableComponentInterface {
+    /**
+     * Currently editing IDP.
+     */
+    editingIDP: IdentityProviderInterface;
+    /**
+    * Show sign on methods condition
+    */
+    isSetStrongerAuth?: boolean;
+    /**
+     * On list item select callback.
+     */
+    onListItemClick?: (event: SyntheticEvent, app: ApplicationListItemInterface) => void;
+    /**
+     * Enable selection styles.
+     */
+    selection?: boolean;
+    /**
+     * Show list item actions.
+     */
+    showListItemActions?: boolean;
+    /**
+     * Default list item limit.
+     */
+    defaultListItemLimit?: number;
+     /**
+     * Is the connected apps info request loading.
+     */
+    isLoading?: boolean;
+    /**
+     * Is the list rendered on a portal.
+     */
+    isRenderedOnPortal?: boolean;
+    /**
+     * Specifies if the component should only be read-only.
+     */
+    isReadOnly: boolean;
+    /**
+     * Loading Component.
+     */
+    loader: () => ReactElement;
```

Comment:
Why are we passing a loader as a prop? Can't we implement that within the component itself?

Reply:
Normally loader is passed to all the tab components in the `identity-provider edit`